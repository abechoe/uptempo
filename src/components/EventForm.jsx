import Radium, { Style } from 'radium';
import React from 'react';
import Input from './Input';
import Checkbox from './Checkbox';
import Button from './Button';

class EventForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      event_start: '',
      event_end: '',
      recurs: false,
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.updateField = this.updateField.bind(this);
  }

  updateField(newState) {
    this.setState(newState);
  }

  handleSubmit(e) {
    e.preventDefault();
    fetch('http://localhost:3000/events', {
      method: 'POST',
      body: JSON.stringify(this.state),
      headers: {
        'Content-Type': 'application/json',
      },
    }).then(function(response) {
      return response.json();
    }).then(function(myJson) {
      console.log(JSON.stringify(myJson));
    });
  }

  render() {
    return (
      <form method="post" onSubmit={this.handleSubmit} style={[styles.base]}>
        <fieldset style={[styles.fieldset]}>
          <h1>Create an Event</h1>
          <Input type="text" id="name" name="name" title="Name" value={this.state.name} onChange={this.updateField} />
          <Input type="datetime-local" id="event_start" name="event_start" title="Start Time" onChange={this.updateField} />
          <Input type="datetime-local" id="event_end" name="event_end" title="End Time" onChange={this.updateField} />
          <Checkbox id="recurs" name="recurs" title="Recurs" onChange={this.updateField} />
          <Button type="submit" title="Submit" />
        </fieldset>
      </form>
    );
  }
}

const styles = {
  base: {
    width: '800px',
    margin: '0 auto',
    font: '12px Helvetica, sans-serif',
    background: '#2f723e',
    color: '#fff',
    borderRadius: '5px',
  },

  fieldset: {
    border: 'none',
  },
};

export default Radium(EventForm);
