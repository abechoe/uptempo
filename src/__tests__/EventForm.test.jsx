import React from 'react';
import { shallow, mount } from 'enzyme';

import EventForm from 'components/EventForm';

describe('EventForm component tests', () => {
  it('renders an EventForm component', () => {
    const wrapper = shallow(<EventForm />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('EventForm submit tests', () => {
  let mockEvent;
  let mockInputValues;
  let mockCreatedEvent;

  beforeEach(() => {
    mockEvent = { preventDefault: jest.fn() };
    mockInputValues = {
      name: 'Sunday Service',
      event_start: '2018-09-06T02:00',
      event_end: '2018-09-06T03:00',
      recurs: true,
      event_interval_join_attributes: {
        interval_attributes: {
          seconds: '30',
        },
      },
      _radiumStyleState: {},
    };

    mockCreatedEvent = {
      "name":"Sunday Service",
      "event_start":"2018-09-06 02:00:00 UTC",
      "event_end":"2018-09-06 03:00:00 UTC",
      "recurs":true,
      "interval":null,
      "tasks":[]
    };

    global.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        event: mockCreatedEvent,
      })
    }));
  });

  it('calls fetch with the supplied input values', () => {
    const wrapper = shallow(<EventForm />);
    const expectedFetchBody = {
      method: 'POST',
      body: JSON.stringify(mockInputValues),
      headers: {
        'Content-Type': 'application/json',
      },
    }

    wrapper.setState(mockInputValues);
    wrapper.instance().handleSubmit(mockEvent);
    expect(global.fetch)
      .toHaveBeenCalledWith(`${process.env.REACT_APP_API_URL}/events`, expectedFetchBody);
  });
});