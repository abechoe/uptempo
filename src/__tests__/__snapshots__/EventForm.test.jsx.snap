// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm component tests renders an EventForm component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <fieldset
        data-radium={true}
        style={
          Object {
            "border": "none",
          }
        }
      >
        <h1>
          Create an Event
        </h1>
        <Input
          id="name"
          name="name"
          onChange={[Function]}
          title="Name"
          type="text"
          value=""
        />
        <Input
          id="event_start"
          name="event_start"
          onChange={[Function]}
          title="Start Time"
          type="datetime-local"
        />
        <Input
          id="event_end"
          name="event_end"
          onChange={[Function]}
          title="End Time"
          type="datetime-local"
        />
        <Checkbox
          id="recurs"
          name="recurs"
          onChange={[Function]}
          title="Recurs"
        />
        <Button
          title="Submit"
          type="submit"
        />
      </fieldset>,
      "data-radium": true,
      "method": "post",
      "onSubmit": [Function],
      "style": Object {
        "background": "#2f723e",
        "borderRadius": "5px",
        "color": "#fff",
        "font": "12px Helvetica, sans-serif",
        "margin": "0 auto",
        "width": "800px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Create an Event
          </h1>,
          <Input
            id="name"
            name="name"
            onChange={[Function]}
            title="Name"
            type="text"
            value=""
          />,
          <Input
            id="event_start"
            name="event_start"
            onChange={[Function]}
            title="Start Time"
            type="datetime-local"
          />,
          <Input
            id="event_end"
            name="event_end"
            onChange={[Function]}
            title="End Time"
            type="datetime-local"
          />,
          <Checkbox
            id="recurs"
            name="recurs"
            onChange={[Function]}
            title="Recurs"
          />,
          <Button
            title="Submit"
            type="submit"
          />,
        ],
        "data-radium": true,
        "style": Object {
          "border": "none",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "host",
          "props": Object {
            "children": "Create an Event",
          },
          "ref": null,
          "rendered": "Create an Event",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "class",
          "props": Object {
            "id": "name",
            "name": "name",
            "onChange": [Function],
            "title": "Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "class",
          "props": Object {
            "id": "event_start",
            "name": "event_start",
            "onChange": [Function],
            "title": "Start Time",
            "type": "datetime-local",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".3",
          "nodeType": "class",
          "props": Object {
            "id": "event_end",
            "name": "event_end",
            "onChange": [Function],
            "title": "End Time",
            "type": "datetime-local",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".4",
          "nodeType": "class",
          "props": Object {
            "id": "recurs",
            "name": "recurs",
            "onChange": [Function],
            "title": "Recurs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".5",
          "nodeType": "class",
          "props": Object {
            "title": "Submit",
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "fieldset",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <fieldset
          data-radium={true}
          style={
            Object {
              "border": "none",
            }
          }
        >
          <h1>
            Create an Event
          </h1>
          <Input
            id="name"
            name="name"
            onChange={[Function]}
            title="Name"
            type="text"
            value=""
          />
          <Input
            id="event_start"
            name="event_start"
            onChange={[Function]}
            title="Start Time"
            type="datetime-local"
          />
          <Input
            id="event_end"
            name="event_end"
            onChange={[Function]}
            title="End Time"
            type="datetime-local"
          />
          <Checkbox
            id="recurs"
            name="recurs"
            onChange={[Function]}
            title="Recurs"
          />
          <Button
            title="Submit"
            type="submit"
          />
        </fieldset>,
        "data-radium": true,
        "method": "post",
        "onSubmit": [Function],
        "style": Object {
          "background": "#2f723e",
          "borderRadius": "5px",
          "color": "#fff",
          "font": "12px Helvetica, sans-serif",
          "margin": "0 auto",
          "width": "800px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Create an Event
            </h1>,
            <Input
              id="name"
              name="name"
              onChange={[Function]}
              title="Name"
              type="text"
              value=""
            />,
            <Input
              id="event_start"
              name="event_start"
              onChange={[Function]}
              title="Start Time"
              type="datetime-local"
            />,
            <Input
              id="event_end"
              name="event_end"
              onChange={[Function]}
              title="End Time"
              type="datetime-local"
            />,
            <Checkbox
              id="recurs"
              name="recurs"
              onChange={[Function]}
              title="Recurs"
            />,
            <Button
              title="Submit"
              type="submit"
            />,
          ],
          "data-radium": true,
          "style": Object {
            "border": "none",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": ".0",
            "nodeType": "host",
            "props": Object {
              "children": "Create an Event",
            },
            "ref": null,
            "rendered": "Create an Event",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": ".1",
            "nodeType": "class",
            "props": Object {
              "id": "name",
              "name": "name",
              "onChange": [Function],
              "title": "Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".2",
            "nodeType": "class",
            "props": Object {
              "id": "event_start",
              "name": "event_start",
              "onChange": [Function],
              "title": "Start Time",
              "type": "datetime-local",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".3",
            "nodeType": "class",
            "props": Object {
              "id": "event_end",
              "name": "event_end",
              "onChange": [Function],
              "title": "End Time",
              "type": "datetime-local",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".4",
            "nodeType": "class",
            "props": Object {
              "id": "recurs",
              "name": "recurs",
              "onChange": [Function],
              "title": "Recurs",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".5",
            "nodeType": "class",
            "props": Object {
              "title": "Submit",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "fieldset",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
